data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java

```diff
@@ -798,6 +798,29 @@ public void filteredValidatorsList_shouldFilterByValidatorPubkey() {
     assertThat(pubkeys).containsExactly(key);
   }
 
+  @Test
+  public void validatorParameterToIndex_shouldThrowBadRequestExceptionWhenIndexInvalid() {
+    final ChainDataProvider provider =
+        new ChainDataProvider(recentChainData, combinedChainDataClient);
+    assertThrows(BadRequestException.class, () -> provider.validatorParameterToIndex("a"));
+  }
+
+  @Test
+  public void validatorParameterToIndex_shouldThrowBadRequestExceptionWhenIndexTooHigh() {
+    final ChainDataProvider provider =
+        new ChainDataProvider(recentChainData, combinedChainDataClient);
+    assertThrows(BadRequestException.class, () -> provider.validatorParameterToIndex("1024000"));
+  }
+
+  @Test
+  public void validatorParameterToIndex_shouldThrowBadRequestExceptionWhenKeyNotFound() {
```

Comment:
We throw when the key isn't found or only when it's an invalid key?

Reply:
key not found is basically search by root, index not found is basically index out of bounds...